invenio_communities/members/services/service.py

```diff
@@ -650,16 +665,12 @@ def rebuild_index(self, identity, uow=None):
 
         Note: Skips (soft) deleted records.
         """
-        for rec_meta in self.record_cls.model_cls.query.all():
-            rec = self.record_cls(rec_meta.data, model=rec_meta)
-
-            if not rec.is_deleted:
-                self.indexer.index(rec)
-
-        for rec_meta in ArchivedInvitation.model_cls.query.all():
-            rec = ArchivedInvitation(rec_meta.data, model=rec_meta)
+        members = self.record_cls.model_cls.query.filter_by(is_deleted=False).all()
```

Comment:
same old question about removing the `if`. Feel free to answer to it in only one of the PRs 😅